src/value/mod.rs

```diff
@@ -318,6 +388,44 @@ pub(crate) fn str_to_c_str(string: &str) -> *const std::os::raw::c_char {
     unsafe { (*c_str).as_ptr() }
 }
 
+pub(crate) fn naive_date_to_mg_date(input: &NaiveDate) -> *mut bindings::mg_date {
+    let unix_epoch = NaiveDate::from_ymd(1970, 1, 1).num_days_from_ce();
+    unsafe { bindings::mg_date_make((input.num_days_from_ce() - unix_epoch) as i64) }
+}
+
+pub(crate) fn naive_local_time_to_mg_local_time(input: &NaiveTime) -> *mut bindings::mg_local_time {
+    let hours_ns = hours_as_seconds(input.hour() as i64) * NSEC_IN_SEC;
+    let minutes_ns = minutes_as_seconds(input.minute() as i64) * NSEC_IN_SEC;
+    let seconds_ns = (input.second() as i64) * NSEC_IN_SEC;
+    let nanoseconds = input.nanosecond() as i64;
+    unsafe { bindings::mg_local_time_make(hours_ns + minutes_ns + seconds_ns + nanoseconds) }
+}
+
+pub(crate) fn naive_local_date_time_to_mg_local_date_time(
+    input: &NaiveDateTime,
+) -> *mut bindings::mg_local_date_time {
+    let unix_epoch = NaiveDate::from_ymd(1970, 1, 1).num_days_from_ce();
+    let days_s = days_as_seconds((input.num_days_from_ce() - unix_epoch) as i64);
+    let hours_s = hours_as_seconds(input.hour() as i64);
+    let minutes_s = minutes_as_seconds(input.minute() as i64);
+    let seconds_s = input.second() as i64;
+    let nanoseconds = input.nanosecond() as i64;
+    unsafe {
+        bindings::mg_local_date_time_make(days_s + hours_s + minutes_s + seconds_s, nanoseconds)
+    }
+}
+
+pub(crate) fn duration_to_mg_duration(input: &Duration) -> *mut bindings::mg_duration {
+    // Duration returns total number of nanoseconds, in order to create a valid mg_duration object,
+    // days and seconds have to be reducted from the total number of nanoseconds.
+    let mut nanoseconds = input.num_nanoseconds().unwrap();
```

Comment:
I think this will over flow pretty "quickly", with [292 years](https://www.google.com/search?q=2%5E63+nanoseconds+in+year&oq=2%5E63+nanoseconds+in+year&aqs=chrome.0.69i59.11560j0j1&sourceid=chrome&ie=UTF-8). It would be better to get some bigger unit and the rest in mili/nanoseconds.